tasks/main.yml

```diff
@@ -245,3 +243,34 @@
   when:
     - ansible_service_mgr == 'systemd'
     - not (keepalived_systemd_overrides | bool)
+
+# Until  https://github.com/ansible/ansible/pull/72337 is fixed, we need to not use
+# the service/systemd module.
+#- name: Ensuring keepalived is enabled
+#  service:
+#    name: "{{ keepalived_service_name }}"
+#    enabled: "yes"
+#    masked: "{{ (ansible_service_mgr is defined and ansible_service_mgr == 'systemd') | ternary('no', omit) }}"
+#
+
+# Reload systemd is necessary if new packages, or changes of overrides.
+# Because there is no notify on those tasks, and that the notify would be happening after the
+# task to ensure keepalived is enabled, move it to an unconditional task here.
+# When the bug above is fixed with the service/systemd module, ensure everything is
+# done in a single handler, tackling the daemon-reloads and the restart of the service,
+# plus a single task, handling the daemon-reload and ensuring the service is enabled
+# on the first install (the changes in systemd files would have to trigger the
+# handler to restart the service).
+- name: Check if keepalived is enabled
+  shell: "systemctl daemon-reload && systemctl is-enabled {{ keepalived_service_name }}"
+  register: isenabled
+  changed_when: isenabled.stdout == "disabled" and isenabled.rc != 0
+  failed_when: isenabled.stdout != "disabled" and isenabled.rc != 0
+  when: ansible_service_mgr == 'systemd'
```

Comment:
I think we won't have cases where service_mgr is not systemd. IIRC you've already dropped these conditionals before

Reply:
I have indeed dropped those in a different branch. My CI is/was testing ubuntu 14, so removing this breaks. I intend to refresh CI and remove those conditionals after this patch.

Reply:
TBH this has introduced a gap on old debian based systems where we don't ensure keepalived has been enabled.  I should fix that.

Reply:
To be clear: I want this to be the latest release for 14.04, then I bump the versions to a new major to highlight the drop of support of upstart.

Reply:
Ah, ok then. It's just not supported for a year and a half, so I thought nobody cares about it:)

Reply:
I do have some people who just contacted me this morning about where this role is heading :facepalm:

Reply:
> Also I thought this bug occurs only with modern kernels (like 5.8) which 14.04 will never have?

So true. I am at risk of introducing bugs, for somethign that doesn't really matter for 14.04.

Reply:
See also https://github.com/evrardjp/ansible-keepalived/issues/160 on why